crates/db_schema/src/impls/person.rs

```diff
@@ -326,6 +327,39 @@ impl Blockable for PersonActions {
   }
 }
 
+impl Notable for PersonActions {
+  type Form = PersonNoteForm;
+  type ObjectIdType = PersonId;
+  type ObjectType = Person;
+
+  async fn note(pool: &mut DbPool<'_>, form: &Self::Form) -> LemmyResult<Self> {
+    let conn = &mut get_conn(pool).await?;
+    insert_into(person_actions::table)
+      .values(form)
+      .on_conflict((person_actions::person_id, person_actions::target_id))
+      .do_update()
+      .set(form)
+      .returning(Self::as_select())
+      .get_result::<Self>(conn)
+      .await
+      .with_lemmy_type(LemmyErrorType::PersonNoteAlreadyExists)
```

Comment:
The error doesnt make sense as it overwrites if note already exists, so theres no error.